src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -162,6 +171,36 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         }
     }
 
+    @computed
+    get genericAssayDataOptionsByGenericAssayType(): {
+        [genericAssayType: string]: ChartOption[];
+    } {
+        const groupedChartMetaByGenericAssayType = _.groupBy(
+            this.groupedChartMetaByDataType[
+                ChartMetaDataTypeEnum.GENERIC_ASSAY
+            ] || [],
+            chartMeta => chartMeta.genericAssayType
+        );
+
+        return _.mapValues(groupedChartMetaByGenericAssayType, chartMeta => {
+            let genericAssayDataOptions = getOptionsByChartMetaDataType(
+                chartMeta,
+                this.selectedAttrs,
+                this.props.store.chartsType.toJS()
+            );
+            if (
```

Comment:
it's not a good idea to hide a check like this inside of a computed like this.    it hides the business logic.  also, the name of this computed doesn't seems to fully describe it.  it seems it is actually returning chart types, right?  if that's true, chart types should be in the name.   then we can have another layer of computeds that name the tabs and which do the filtering.   e.g.   chartTypeOptionsForClinicalDataTab.

Reply:
It's actually returns the `chartOptions for generic assay`, so rename to `genericAssayChartOptionsByGenericAssayType`